### PR TITLE
Add an accessor to retrieve the dimension of SolutionUserObject

### LINKED ISSUE #10139: Add an accessor to retrieve the dimension of SolutionUserObject
## Rationale

We have created a `Function` constructing a `SolutionUserObject` and we would like to check that the `MeshFile` read by the `SolutionUserObject` has the same dimension as the mesh for the current simulation.

## Description

If this change is approved, I would simply add a public accessor to `SolutionUserObject` returning `_mesh->spatial_dimension()`.

## Impact

This will help enforce that the mesh dimensions are consistent.

### PR DESCRIPTION
**Pull request by vincentlaboure**
closes #10139

### CODE REVIEW

**Comment by andrsd** [comment]
Before merging this, let me check that this works with RELAP-7. We do not have a test case that would be using SolutionFunction + SolutionUserObject, but we have to be able to take 2D rattlesnake solution and use it in our setup where we have 1D and 2D elements. We claim our meshes are actually 3D (because they actual are).  We are "transferring" the 2D solution read from a file onto our 2D heat structure elements, which kosher, but I think this check would break this.

**Comment by vincentlaboure** [comment]
I see your point: I can remove that check and have it only in our own `Function` in Rattlesnake

**Comment by andrsd** [comment]
So, this breaks RELAP-7. I built a test for this and when it gets in the RELAP-7's devel, we can re-run the tests to see it.

File: framework/include/userobject/SolutionUserObject.h [hunk 1]
```diff
@@ -221,6 +221,9 @@ class SolutionUserObject : public GeneralUserObject
                      "found_first");
   }
 
+  /// Return the spatial dimension of the mesh file
```
Comment (andrsd) [inline framework/include/userobject/SolutionUserObject.h]: This should be `/** ... */` with `@return` etc.

**Review by andrsd — CHANGES_REQUESTED** [review]

**Comment by YaqiWang** [comment]
Is this ready? There is a MR in Rattlesnake depending on this.

**Comment by andrsd** [comment]
Yeah, it is fine now. Thanks for the ping @YaqiWang

**Review by andrsd — APPROVED** [review]

**Pull request merged**